modules/ui/feature_list.js

```diff
@@ -122,14 +122,16 @@ export function uiFeatureList(context) {
 
             if (!q) return result;
 
-            var idMatch = q.match(/^([nwr])([0-9]+)$/);
+            var idMatch = q.match(/^(.*[^\w])?(node|n|way|w|relation|r)[^\w]?([0-9]+)([^\w].*)?$/i);
 
             if (idMatch) {
+                var elemType = idMatch[2][0];
+                var elemId = idMatch[3];
```

Comment:
If you were to change to my simplified regex as suggested these two indexes both drop 1 (so this becomes)
```suggestion
                var elemId = idMatch[2];
```
and the other becomes 1.